$NetBSD: patch-ab,v 1.1 2006/01/13 07:51:42 rillig Exp $

This patch avoid tons of #line warnings when parsing the output of newer
gcc.

--- src/c-lex.c.orig	2003-09-10 20:52:02.000000000 +0200
+++ src/c-lex.c	2006-01-12 03:05:04.000000000 +0100
@@ -843,6 +843,9 @@ linenum:
 	  c = GETC();
 	  while (c == ' ' || c == '\t')
 	    c = GETC();
+	  /* Skip the file type (3 = system header, ...) of gcc */
+	  if (isdigit(c))
+	    c = GETC();
 	  if (c == '\n')
 	    return c;
 	  UNGETC (c);
